### PR TITLE
 Allow node content to be of type number.

### PR DESCRIPTION
**Pull request by nikeshmhr**
This PR fixes issue https://github.com/parsable/react-truncate-markup/issues/88

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ const cloneWithChildren = (node, children, isRootEl, level) => {
 };
 
 const validateTree = (node) => {
-  if (typeof node === 'string' || isAtomComponent(node)) {
+  if (['string', 'number'].includes(typeof node) || isAtomComponent(node)) {
```
Comment (patrik-piskay) [inline src/index.js:59]: I realized we need to check for other values here. let's add a null/undefined check as the first thing, like
```js
if (node == null || ...)
```

and at the end, when checking for `node.props.children`, let's be a bit more defensive, and check for `node.props` first, like
```js
if (node.props && node.props.children)
```

**Review by patrik-piskay — COMMENTED** [review]
Would you mind creating a test for this use case? A new test case like [this one](https://github.com/parsable/react-truncate-markup/blob/master/__tests__/browser.js#L205-L218) would suffice, I imagine. Just testing with numbers, like in your bug report.

File: src/index.js (hunk 1 above)
Comment (nikeshmhr) [reply src/index.js]: I've added a test case and checks. I'm not sure if we need to check `null/undefined`.

Comment (patrik-piskay) [reply src/index.js]: nit: checking for `node == null` will take care of both the null and undefined checks

File: src/index.js [hunk 2]
```diff
@@ -56,7 +56,12 @@ const cloneWithChildren = (node, children, isRootEl, level) => {
 };
 
 const validateTree = (node) => {
-  if (typeof node === 'string' || isAtomComponent(node)) {
+  if (
+    node === null ||
```
Comment (patrik-piskay) [inline src/index.js:60]: nit: checking for node == null will take care of both the null and undefined checks

Comment (nikeshmhr) [reply src/index.js:60]: Updated.

Comment (patrik-piskay) [reply src/index.js:60]: it should be `==`, not `===` 😅  Can you fix this and I'll merge it after?

Comment (nikeshmhr) [reply src/index.js]: Done

**Pull request merged**